### PR TITLE
Adding ability to use email in URL

### PR DESCRIPTION
**Pull request by Dillon-Dumesnil**
We already support searching by email or username,
but the URL only supported username. This allows the URL
to use either username or email.

This PR also has various refactors to make the code more DRY

### CODE REVIEW

File: src/users/UserPage.jsx [hunk 1]
```diff
@@ -26,12 +30,21 @@ export default function UserPage({ match }) {
...
+    if (userIdentifier) {
+      if (isEmail(userIdentifier)) {
+        history.replace(`/users/?email=${userIdentifier}`);
+      } else if (isValidUsername(userIdentifier)) {
+        history.replace(`/users/?username=${userIdentifier}`);
+      }
     }
-    if (result.errors.length > 0) {
-      result.errors.forEach(error => add(error));
+    console.log(errors);
```
Comment (davidjoy) [inline src/users/UserPage.jsx]: Can probably remove this console log if we're adding them to the alert list.

Comment (Dillon-Dumesnil) [reply src/users/UserPage.jsx]: That would have been my debugging and I thought I removed it. My bad. One sec

**Comment by davidjoy** [comment]
This seems nice!  

One thing I _think_ you might be able to do is to use the `username` in `data`, rather than even having the username useState hook.  Whenever we search, `data` gets set and should include it, and we should be able to then just pass it down to the sub-components that need it.

Might be worth 5 minutes to see if that's possible - if it's hairy, then I think it's fine to leave.

Cool!

**Review by davidjoy — APPROVED** [review]
See one comment about possibly being able to get rid of `username` state.

**Review by staubina — APPROVED** [review]
I am not seeing anything blocking here.  I recommend we all test on Stage though.

**Comment by Dillon-Dumesnil** [comment]
> This seems nice!
> 
> One thing I _think_ you might be able to do is to use the `username` in `data`, rather than even having the username useState hook. Whenever we search, `data` gets set and should include it, and we should be able to then just pass it down to the sub-components that need it.
> 
> Might be worth 5 minutes to see if that's possible - if it's hairy, then I think it's fine to leave.
> 
> Cool!

Wonderful callout. You are absolutely correct. Removed the set username logic in favor of data.user.username!

**Review by staubina — APPROVED** [review]

**Pull request merged**